FileIO/VtkIO/PVDFile.cpp

```diff
@@ -17,30 +17,27 @@
 namespace FileIO
 {
 
-PVDFile::PVDFile(const std::string &fn)
-    : _fh{fn.c_str()}
+void PVDFile::addVTUFile(const std::string &vtu_fname, double timestep)
 {
-    if (!_fh) {
-        ERR("could not open file `%s'", fn.c_str());
+    _datasets.push_back(std::make_pair(timestep, vtu_fname));
+
+    std::ofstream fh(_pvd_filename.c_str());
+    if (!fh) {
+        ERR("could not open file `%s'", vtu_fname.c_str());
         std::abort();
     }
 
-    _fh << std::setprecision(std::numeric_limits<double>::digits10);
+    fh << std::setprecision(std::numeric_limits<double>::digits10);
 
-    _fh << "<?xml version=\"1.0\"?>\n"
+    fh << "<?xml version=\"1.0\"?>\n"
```

Comment:
I don't really like that now you rewrite the PVD file after every output timestep.
But since you want to work with the PVD file while the simulation is still running, that's the easiest way to accomplish that.
I don't have a better idea.